helm/esdk/templates/huawei-csi-node.yaml

```diff
@@ -0,0 +1,137 @@
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: huawei-csi-node
+  namespace: kube-system
+
+---
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: huawei-csi-driver-registrar-runner
+  namespace: kube-system
+rules:
+  - apiGroups: [""]
+    resources: ["events"]
+    verbs: ["get", "list", "watch", "create", "update", "patch"]
+
+---
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: huawei-csi-driver-registrar-role
+subjects:
+  - kind: ServiceAccount
+    name: huawei-csi-node
+    namespace: kube-system
+roleRef:
+  kind: ClusterRole
+  name: huawei-csi-driver-registrar-runner
+  apiGroup: rbac.authorization.k8s.io
+---
+kind: DaemonSet
+apiVersion: apps/v1
+metadata:
+  name: huawei-csi-node
+  namespace: kube-system
+spec:
+  selector:
+    matchLabels:
+      app: huawei-csi-node
+  template:
+    metadata:
+      labels:
+        app: huawei-csi-node
+    spec:
+      serviceAccountName: huawei-csi-node
+      hostPID: true
+      hostNetwork: true
+      containers:
+        - name: csi-node-driver-registrar
+          image: {{ .Values.images.sidecar.registrar }}
+          args:
+            - "--csi-address=/csi/csi.sock"
+            - "--kubelet-registration-path=/var/lib/kubelet/plugins/csi.huawei.com/csi.sock"
+          imagePullPolicy: "IfNotPresent"
+          volumeMounts:
+            - name: plugin-dir
+              mountPath: /csi
+            - name: registration-dir
+              mountPath: /registration
+        - name: huawei-csi-driver
+          image: {{ required "Must provide the CSI controller service node image." .Values.images.huaweiCsiNodeService }}
+          args:
+            - "--endpoint=/csi/csi.sock"
+            - "--containerized"
+            - "--driver-name=csi.huawei.com"
+            - "--volume-use-multipath=true"
+          securityContext:
+            privileged: true
+            capabilities:
+              add: ["SYS_ADMIN"]
+            allowPrivilegeEscalation: true
+          imagePullPolicy: "IfNotPresent"
+          lifecycle:
+            preStop:
+              exec:
+                command: ["/bin/sh", "-c", "rm -f /csi/csi.sock"]
+          env:
+            - name: CSI_ESDK_NODENAME
+              valueFrom:
+                fieldRef:
+                  apiVersion: v1
+                  fieldPath: spec.nodeName
```

Comment:
IMO, Currently the "CSI_ESDK_NODENAME" is not used. Please remove it

Reply:
Should I leave the `env:` section empty or remove the section altogether?

Reply:
I have removed the env variable `CSI_ESDK_NODENAME` for now and retained the `env:` section.

Reply:
Please remove ```env``` section as well

Reply:
Done.